subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformedFileCache.java

```diff
@@ -224,4 +266,28 @@ public void execute(File outputDir) {
             result = ImmutableList.copyOf(transformer.apply(inputFile, outputDir));
         }
     }
+
+    private class TransformedFileCacheCleanup extends FixedAgeOldestCacheCleanup {
```

Comment:
⭕️ We should consider cleaning up empty folders at some point.

Reply:
Agreed, I'll add it to the list of things to address after merging this PR. 🙂